pkg/endpointmanager/manager.go

```diff
@@ -460,6 +464,35 @@ func (mgr *EndpointManager) AddEndpoint(owner regeneration.Owner, ep *endpoint.E
 	return nil
 }
 
+func (mgr *EndpointManager) AddHostEndpoint(ctx context.Context, owner regeneration.Owner,
+	proxy endpoint.EndpointProxy, allocator cache.IdentityAllocator, reason string, nodeName string) (err error) {
+	ep := endpoint.CreateHostEndpoint(owner, proxy, allocator)
+	if err := mgr.AddEndpoint(owner, ep, reason); err != nil {
+		return err
+	}
+
+	epLabels := labels.Labels{}
+	epLabels.MergeLabels(labels.LabelHost)
+
+	if k8s.IsEnabled() {
+		// Retrieve k8s labels.
+		if k8sNode, err := k8s.GetNode(k8s.Client(), nodeName); err != nil {
+			log.WithError(err).Warning("Kubernetes node resource representing own node is not available, cannot set Labels")
+		} else {
+			newLabels := labels.Map2Labels(k8sNode.GetLabels(), labels.LabelSourceK8s)
+			newIdtyLabels, _ := labelsfilter.Filter(newLabels)
```

Comment:
Not sure whether we need special handling for node labels, eg whether the pod labels filter will be appropriate for filtering node labels. Can likely be followed up separately. /cc @aanm

Reply:
@joestringer If I understood your question, I think Paul added a k8s watcher for nodes to handle this

Reply:
My concern was that we have `pkg/labelsfilter/` which was built for filtering labels that may apply to pods. Now that we will filter nodes labels as well, there could be additional prefixes we need to ignore / respect.

Reply:
Looking at `pkg/labelsfilter/filter.go`, I think we do need to update the filtering logic. Maybe updating the list of expressions would be enough though? Unless we need separate lists for pods and nodes? https://github.com/cilium/cilium/blob/ebe156fd7ce87cadaf4cacf3b1cfe151bf9a51d4/pkg/labelsfilter/filter.go#L162-L174

Reply:
Perhaps it would be good to have a dedicated filter for nodes. These are the current labels I see on my nodes:
```
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/os: linux
    kubernetes.io/arch: amd64
    kubernetes.io/hostname: k8s1
    kubernetes.io/os: linux
```

Reply:
I've removed this commit from the PR and we'll submit it as a follow up with the required changes for the label filtering.